jb/itch5/array_based_order_book.hpp

```diff
@@ -0,0 +1,660 @@
+#ifndef jb_itch5_array_based_order_book_hpp
+#define jb_itch5_array_based_order_book_hpp
+
+#include <jb/itch5/half_quote.hpp>
+#include <jb/itch5/price_field.hpp>
+#include <jb/itch5/price_levels.hpp>
+#include <jb/config_object.hpp>
+#include <jb/feed_error.hpp>
+#include <jb/log.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <map>
+#include <sstream>
+#include <utility>
+#include <vector>
+
+namespace jb {
+namespace itch5 {
+
+namespace defaults {
+
+#ifndef JB_ARRAY_DEFAULTS_max_size
+#define JB_ARRAY_DEFAULTS_max_size 10000
+#endif
+
+std::size_t max_size = JB_ARRAY_DEFAULTS_max_size;
+
+} // namespace defaults
+
+template <typename compare_t>
+class array_based_book_side;
+
+/**
+ * Define the types of buy and sell side classes.
+ *
+ * It is used as template parameter book_type of the
+ * template classes order_book and compute_book:
+ * - usage: jb::itch5::order_book<jb::itch5::array_based_order_book>
+ */
+struct array_based_order_book {
+  using buys_t = array_based_book_side<std::greater<price4_t>>;
+  using sells_t = array_based_book_side<std::less<price4_t>>;
+  class config;
+};
+
+/**
+ * Configure an array_based_order_book config object
+ */
+class array_based_order_book::config : public jb::config_object {
+public:
+  config()
+      : max_size(
+            desc("max-size")
+                .help(
+                    "Configure the max size of a array based order book."
+                    " Only used when enable-array-based is set"),
+            this, jb::itch5::defaults::max_size) {
+  }
+
+  config_object_constructors(config);
+
+  /// Validate the configuration
+  void validate() const override {
+    if ((max_size() <= 0) or (max_size() > jb::itch5::defaults::max_size)) {
+      std::ostringstream os;
+      os << "max-size must be > 0 and <=" << jb::itch5::defaults::max_size
+         << ", value=" << max_size();
+      throw jb::usage(os.str(), 1);
+    }
+  }
+
+  jb::config_attribute<config, std::size_t> max_size;
+};
+
+/// price level limit between mill and penny
+int constexpr TK_DOLLAR = 10000;
```

Comment:
ACK.

Done, just because I am embedded on Chrismas mood.
Create a static member of price_field class called dollar_price.
I know you are the "good names" person so just let me know the correct one and I will change it